server/types/oid/regtype.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oid
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/lib/pq/oid"
+
+	"github.com/dolthub/doltgresql/postgres/parser/types"
+	pgtypes "github.com/dolthub/doltgresql/server/types"
+)
+
+// regtype_IoInput is the implementation for IoInput that avoids circular dependencies by being declared in a separate
+// package.
+func regtype_IoInput(ctx *sql.Context, input string) (uint32, error) {
+	// If the string just represents a number, then we return it.
+	if parsedOid, err := strconv.ParseUint(input, 10, 32); err == nil {
+		return uint32(parsedOid), nil
+	}
+	sections, err := ioInputSections(input)
+	if err != nil {
+		return 0, err
+	}
+	if err = regtype_IoInputValidation(ctx, input, sections); err != nil {
+		return 0, err
+	}
+	var relationName string
+	switch len(sections) {
+	case 1:
+		relationName = sections[0]
+	case 3:
+		relationName = sections[2]
+	default:
+		return 0, fmt.Errorf("regtype failed validation")
+	}
+	relationName = normalizeTypeRelationName(relationName)
```

Comment:
Small nitpick, but we should call this `typeName`. Relation name is more for the names that `regclass` represents.